src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,704 @@
+
+        return target_page_id
+
+    def _get_intent_id_from_user_input(
+        self,
+        user_input: types.ConversationTurn.UserInput,
+        source: dict,
+        target: dict) -> Union[str, None]:
+        """Attempt to find the intent id from the source's agent
+          conversation turn user input intent id to the target's pages map
+        """
+
+        source_intent_id = user_input.input.intent
+        source_intent = self._get_intent_name_by_intent_id(
+            intent_id=source_intent_id,
+            intents_map=source["intents_map"])
+        target_intent_id = self._get_intent_id_by_intent_name(
+            intent_name=source_intent,
+            intents_map=target["intents_map_reverse"])
+
+        return target_intent_id
+
+    def _get_new_conversation_turns(
+        self,
+        source: dict,
+        target: dict,
+        conversation_turns: types.ConversationTurn
+        ) -> Union[List[types.ConversationTurn], None]:
+        """ Update the uuids of the current page and intent
+            by display name in each conversation turn.
+          Args:
+            source: Source agent's commons_config dictionary
+            target: Target agent's commons_config dictionary
+            conversation_turns: types.ConversationTurn
+          Returns:
+            types.test_case_conversation_turns
+        """
+
+        new_conversation_turns = []
+        for conv_turn in conversation_turns:
+            user_input_intent = conv_turn.user_input
+            virtual_agent_output = conv_turn.virtual_agent_output
+            if user_input_intent.input.intent:
+                conv_turn.user_input.input.intent = (
+                    self._get_intent_id_from_user_input(
+                        user_input=user_input_intent,
+                        source=source, target=target)
+                )
+                if not conv_turn.user_input.input.intent:
+                    return None
+            if virtual_agent_output.current_page:
+                conv_turn.virtual_agent_output.current_page.name = (
+                    self._get_page_id_from_virtual_agent_output(
+                        virtual_agent_output=virtual_agent_output,
+                        source=source, target=target)
+                )
+                if not conv_turn.virtual_agent_output.current_page.name:
+                    return None
+            if virtual_agent_output.triggered_intent:
+                conv_turn.virtual_agent_output.triggered_intent.name = (
+                    self._get_intent_id_from_virtual_agent_output(
+                        virtual_agent_output=virtual_agent_output,
+                        target=target
+                    )
+                )
+                if not conv_turn.virtual_agent_output.triggered_intent.name:
+                    return None
+            new_conversation_turns.append(conv_turn)
+
+        return new_conversation_turns
+
+    def _missing_attribute_logger(
```

Comment:
@jkshj21 let's rename to `_log_missing_attributes` to adhere to the naming styles we're typically using.

i.e.
- `get_something`
- `list_something`
- `log_something`

The general style is `<action>_<thing you are performing action on>`

Reply:
ok, understood. I keep that in mind. thank you!